node/lib/util/status_util.js

```diff
@@ -387,11 +388,13 @@ exports.getRepoStatus = co.wrap(function *(repo, options) {
             const treeId = head.treeId();
             tree = yield NodeGit.Tree.lookup(repo, treeId);
         }
+        repo.submoduleCacheAll();
         const status = yield DiffUtil.getRepoStatus(repo,
                                                     tree,
                                                     options.paths,
                                                     options.ignoreIndex,
                                                     options.showAllUntracked);
+        repo.submoduleCacheClear();
```

Comment:
Yes, I wasn't sure it would be the case (that we would always cache during a single operation), but you're right.